### PR TITLE
Remove code tag for commit hash in issues for backtrace reports

### PR DESCRIPTION
**Pull request by ZehMatt**
Because backtrace wraps the commit in a code tag it can't be clicked, until they start fixing it this is a quick work-around, its quite annoying that I have to manually navigate to the commit. This job only runs if the issue contains the backtrace.io label.

Here is an corrected example issue: https://github.com/ZehMatt/OpenRCT2/issues/39

### CODE REVIEW

**Comment by James103** [comment]
Does this run for all prior (open) `backtrace.io` issues, or only those that are newly published?

**Comment by ZehMatt** [comment]
> Does this run for all prior (open) `backtrace.io` issues, or only those that are newly published?

No just on new ones, didn't bother with the existing ones, that would be way more work.

**Review by tupaschoal — COMMENTED** [review]
Would it fail if backtrace fixed it on their end? If not, perhaps we should make it so, since that'll make us notice when we can disable this.

**Comment by ZehMatt** [comment]
There is currently no failure, its a quite basic bash script using sed and regex to get rid of the code tags, its questionable when and if they even think about changing this. Ideally backtrace lets the user specify exactly what is posted and how rather than just having fields.

File: .github/workflows/backtrace-commit-fix.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      ISSUE: ${{ github.event.issue.html_url }}
+    if: |
+      contains(github.event.issue.labels.*.name, 'backtrace.io')
+    steps:
+      - name: Modify issue contents
+        env:
+          BODY: ${{ github.event.issue.body }}
+        id: fix
+        run: |
+          changed_body=$(echo "$BODY" | sed 's#<code>\([^<]*\)</code>#\1#g')
```
Comment (Broxzier) [inline .github/workflows/backtrace-commit-fix.yml:20]: `<code>\([^<]*\)</code>` may match more than just the hash. I think it's saver to narrow it to `[0-9a-f]{7}` (or possibly `{7,8}` for future longer hashes).

```suggestion
          changed_body=$(echo "$BODY" | sed 's#<code>\([0-9a-f]{7}\)</code>#\1#g')
```

Comment (Broxzier) [reply .github/workflows/backtrace-commit-fix.yml:20]: I see https://github.com/OpenRCT2/OpenRCT2/issues/19501 was changes while checkout out an issue. Did you update this manually?

Comment (ZehMatt) [reply .github/workflows/backtrace-commit-fix.yml:20]: Yea

**Comment by ZehMatt** [comment]
Made some minor changes, the code tag is now only removed from the commit section and in case its unable to modify the contents it will error, this way we know if backtrace changed this on their own.

File: .github/workflows/backtrace-commit-fix.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          BODY: ${{ github.event.issue.body }}
+        id: fix
+        run: |
+          changed_body=$(echo "$BODY" | sed '/<strong>commit<\/strong>/,/<\/li>/ s/\(<code>\)\(.*\)\(<\/code>\)/\2/g')
+          if [[ "$changed_body" == "$BODY" ]]; then
+            echo "Unable to match the Backtrace report format."
+            exit 1
+          fi
+          echo "$changed_body" >> body.txt
+          gh issue edit $ISSUE --body-file body.txt
```
Comment (Broxzier) [inline .github/workflows/backtrace-commit-fix.yml:26]: Reintroduce the missing newling at end of file:
```suggestion
          gh issue edit $ISSUE --body-file body.txt

```

File: .github/workflows/backtrace-commit-fix.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      ISSUE: ${{ github.event.issue.html_url }}
+    if: |
+      contains(github.event.issue.labels.*.name, 'backtrace.io')
+    steps:
+      - name: Modify issue contents
+        env:
+          BODY: ${{ github.event.issue.body }}
+        id: fix
+        run: |
+          changed_body=$(echo "$BODY" | sed '/<strong>commit<\/strong>/,/<\/li>/ s/\(<code>\)\(.*\)\(<\/code>\)/\2/g')
```
Comment (Broxzier) [inline .github/workflows/backtrace-commit-fix.yml:20]: This line can be made a bit shorter by removing the unnecessary groups in the regex, and passing BODY into sed without piping from echo.
```suggestion
          changed_body=$(sed '/<strong>commit<\/strong>/,/<\/li>/ s/<code>\(.*\)<\/code>/\1/' <<< "$BODY")
```

Comment (ZehMatt) [reply .github/workflows/backtrace-commit-fix.yml:20]: While its shorter its also harder to read in my opinion, makes it quite clear that the output is piped and then processed by sed, the suggestion is harder to understand until you reach the end of the line.

Comment (Broxzier) [reply .github/workflows/backtrace-commit-fix.yml:20]:
```suggestion
          changed_body=$(echo "$BODY" | sed '/<strong>commit<\/strong>/,/<\/li>/ s/<code>\([0-9a-f]{7,8}\)<\/code>/\1/')
```

**Review by Broxzier — APPROVED** [review]

**Pull request merged**